preamble-128.js

```diff
@@ -0,0 +1,33 @@
+/* global Decimal */
+
+// Decimal128 allows for 34 digits of significand
+Decimal.set({ precision: 34 });
+
+const _realMathMethods = {};
+
+// unary Math methods that have a corresponding Decimal static method
+["abs", "floor", "log10"].forEach((mathMethod) => {
+  _realMathMethods[mathMethod] = Math[mathMethod];
+  Math[mathMethod] = function (arg) {
+    if (arg instanceof Decimal) return Decimal[mathMethod](arg);
+    return _realMathMethods[mathMethod](arg);
+  };
+});
+
+// Math methods with nargs > 1 that have a corresponding Decimal static method
+["pow"].forEach((mathMethod) => {
+  _realMathMethods[mathMethod] = Math[mathMethod];
+  Math[mathMethod] = function (...args) {
+    let anyDecimals = false;
+    let allDecimals = true;
+    for (const arg of args) {
+      if (arg instanceof Decimal) anyDecimals = true;
+      else allDecimals = false;
+    }
```

Comment:
Rather than writing this in a stateful way, we can use built-in list methods and braces to make it a bit more straightforward / non-state-tracking and remove a place where it's essentially become whitespace significant:

```js
// inside a proxy 

const decimals = args.filter((el) => el instanceof Decimal);

if (decimals.length === 0) {
  return target.apply(thisArg, args);
}

if (decimals.length < args.length) {
  throw new TypeError(`mixed types not supported in Math.${mathMethod}`);
}

return Decimal[method].apply(thisArg, args);

```